lib/reporter.ex

```diff
@@ -0,0 +1,143 @@
+defmodule Membrane.TelemetryMetrics.Reporter do
+  @moduledoc """
+  Attaches handlers to :telemetry events basing on received list of metrics definitions.
+  Attached handlers store metrics values in ETS tables.
+  These values can be get by calling `scrape/2` function or also reseted by calling `scrape_and_cleanup/2`.
+
+  Currently supported types of metrics are:
+   - `Telemetry.Metrics.Counter`
+   - `Telemetry.Metrics.Sum`
+   - `Telemetry.Metrics.LastValue`
+
+  Currently supported fields of metrics definitions are: `:name`, `:event_name`, `measurement`.
+  Fields `:keep`, `:reporter_options`, `tag_values`, `tags`, `:unit` and functionalities related to them are not supported yet.
+  Metrics values are grouped by  value related to key `:telemetry_metadata` in event `metadata`.
+  """
+
+  use GenServer
+
+  alias Membrane.TelemetryMetrics.Reporter.{Counter, LastValue, Sum}
+
+  @type reporter() :: pid() | atom()
+  @type report() :: map()
+
+  @spec start_link(list(), GenServer.options()) :: GenServer.on_start()
+  def start_link(init_arg, options \\ []) do
+    GenServer.start_link(__MODULE__, init_arg, options)
+  end
+
+  @spec scrape(reporter(), non_neg_integer()) :: report()
+  def scrape(reporter, timeout \\ 5000) do
+    GenServer.call(reporter, :scrape, timeout)
+  end
+
+  @spec scrape_and_cleanup(reporter(), non_neg_integer()) :: report()
+  def scrape_and_cleanup(reporter, timeout \\ 5000) do
+    GenServer.call(reporter, :scrape_and_cleanup, timeout)
+  end
+
+  @impl true
+  def init(init_arg) do
+    metrics = init_arg[:metrics]
```

Comment:
It looks like this argument is required, but the access call can get something that's `nil`. I suggest either `Keyword.fetch!` to raise an error if it's missing or `Keyword.get` with a default empty list.